// @connext/types/dist/src/cf/index.js
import { coinBalanceRefundStateEncoding, multiAssetMultiPartyCoinTransferEncoding, multiAssetMultiPartyCoinTransferInterpreterParamsEncoding, OutcomeType, singleAssetTwoPartyCoinTransferInterpreterParamsEncoding, TwoPartyFixedOutcome, twoPartyFixedOutcomeInterpreterParamsEncoding, virtualAppAgreementEncoding } from "./data-types";
import { CFCoreTypes } from "./node";
export const CF_PATH = "m/44'/60'/0'/25446";
export const EXPECTED_CONTRACT_NAMES_IN_NETWORK_CONTEXT = [
    "ChallengeRegistry",
    "ConditionalTransactionDelegateTarget",
    "CoinBalanceRefundApp",
    "MultiAssetMultiPartyCoinTransferInterpreter",
    "IdentityApp",
    "MinimumViableMultisig",
    "ProxyFactory",
    "SingleAssetTwoPartyCoinTransferInterpreter",
    "TimeLockedPassThrough",
    "TwoPartyFixedOutcomeInterpreter",
    "TwoPartyFixedOutcomeFromVirtualAppInterpreter"
];
export { coinBalanceRefundStateEncoding, multiAssetMultiPartyCoinTransferEncoding, multiAssetMultiPartyCoinTransferInterpreterParamsEncoding, singleAssetTwoPartyCoinTransferInterpreterParamsEncoding, CFCoreTypes, OutcomeType, twoPartyFixedOutcomeInterpreterParamsEncoding, TwoPartyFixedOutcome, virtualAppAgreementEncoding };
