// @connext/types/dist/src/cf/node.js
export var CFCoreTypes;
(function (CFCoreTypes) {
    let ErrorType;
    (function (ErrorType) {
        ErrorType["ERROR"] = "error";
    })(ErrorType = CFCoreTypes.ErrorType || (CFCoreTypes.ErrorType = {}));
    let MethodName;
    (function (MethodName) {
        MethodName["ACCEPT_STATE"] = "acceptState";
        MethodName["GET_PROPOSED_APP_INSTANCE"] = "getProposedAppInstance";
    })(MethodName = CFCoreTypes.MethodName || (CFCoreTypes.MethodName = {}));
    CFCoreTypes.RpcMethodNames = {
        chan_create: "chan_create",
        chan_deposit: "chan_deposit",
        chan_deployStateDepositHolder: "chan_deployStateDepositHolder",
        chan_getChannelAddresses: "chan_getChannelAddresses",
        chan_getAppInstance: "chan_getAppInstance",
        chan_getAppInstances: "chan_getAppInstances",
        chan_getStateDepositHolderAddress: "chan_getStateDepositHolderAddress",
        chan_getFreeBalanceState: "chan_getFreeBalanceState",
        chan_getTokenIndexedFreeBalanceStates: "chan_getTokenIndexedFreeBalanceStates",
        chan_getProposedAppInstances: "chan_getProposedAppInstances",
        chan_getState: "chan_getState",
        chan_getStateChannel: "chan_getStateChannel",
        chan_install: "chan_install",
        chan_requestDepositRights: "chan_requestDepositRights",
        chan_installVirtual: "chan_installVirtual",
        chan_proposeInstall: "chan_proposeInstall",
        chan_rejectInstall: "chan_rejectInstall",
        chan_updateState: "chan_updateState",
        chan_takeAction: "chan_takeAction",
        chan_uninstall: "chan_uninstall",
        chan_uninstallVirtual: "chan_uninstallVirtual",
        chan_rescindDepositRights: "chan_rescindDepositRights",
        chan_withdraw: "chan_withdraw",
        chan_withdrawCommitment: "chan_withdrawCommitment"
    };
    CFCoreTypes.EventNames = {
        CREATE_CHANNEL_EVENT: "CREATE_CHANNEL_EVENT",
        DEPOSIT_CONFIRMED_EVENT: "DEPOSIT_CONFIRMED_EVENT",
        DEPOSIT_FAILED_EVENT: "DEPOSIT_FAILED_EVENT",
        DEPOSIT_STARTED_EVENT: "DEPOSIT_STARTED_EVENT",
        INSTALL_EVENT: "INSTALL_EVENT",
        INSTALL_VIRTUAL_EVENT: "INSTALL_VIRTUAL_EVENT",
        REJECT_INSTALL_EVENT: "REJECT_INSTALL_EVENT",
        UNINSTALL_EVENT: "UNINSTALL_EVENT",
        UNINSTALL_VIRTUAL_EVENT: "UNINSTALL_VIRTUAL_EVENT",
        UPDATE_STATE_EVENT: "UPDATE_STATE_EVENT",
        WITHDRAWAL_CONFIRMED_EVENT: "WITHDRAWAL_CONFIRMED_EVENT",
        WITHDRAWAL_FAILED_EVENT: "WITHDRAWAL_FAILED_EVENT",
        WITHDRAWAL_STARTED_EVENT: "WITHDRAWAL_STARTED_EVENT",
        PROPOSE_INSTALL_EVENT: "PROPOSE_INSTALL_EVENT",
        PROTOCOL_MESSAGE_EVENT: "PROTOCOL_MESSAGE_EVENT"
    };
})(CFCoreTypes || (CFCoreTypes = {}));
